pkgs/applications/networking/deck/update.sh

```diff
@@ -0,0 +1,20 @@
+#!/usr/bin/env nix-shell
+#!nix-shell -I nixpkgs=./. -i bash -p curl coreutils jq
+
+set -euo pipefail
+cd "$(dirname "${BASH_SOURCE[0]}")"
+
+DECK_LATEST_VERSION=$(curl -s "https://api.github.com/repos/kong/deck/releases?per_page=1" | jq -r '.[0].name')
+CURRENT_VERSION=v$(sed -nE 's/\s*version = "(.*)".*/\1/p' ./default.nix)
+
+if [[ "$DECK_LATEST_VERSION" == "$CURRENT_VERSION" ]]; then
+  echo "Already up to date!"
+  exit 0
+fi
+
+CURRENT_HASH=$(sed -nE 's/\s*short_hash = "(.*)".*/\1/p' ./default.nix)
+NEW_HASH=$(curl -s https://api.github.com/repos/kong/deck/tags | jq -rc '.[] | select(.name=="$DECK_LATEST_VERSION")|.commit.sha' | head -c 7 -)
+
+sed -i "s|$CURRENT_HASH|$NEW_HASH|" "./default.nix"
+
+update-source-version deck "${DECK_LATEST_VERSION//v/}"
```

Comment:
Running update for:
 - deck-1.25.0: UPDATING ...
 - deck-1.25.0: ERROR

--- SHOWING ERROR LOG FOR deck-1.25.0 ----------------------

/home/joerg/git/nixpkgs/pkgs/applications/networking/deck/update.sh: line 20: update-source-version: command not found

Reply:
This should be fixed now. 

Can you please tell me how you run the test?

I use `nix-shell maintainers/scripts/update.nix --argstr maintainer liyangau` to test but it did not catch this error.

Reply:
I usually do `nix-update --build --commit --use-update-script deck`

Reply:
I also downgraded deck beforehand to actually made it upgrade.

Reply:
It's still failing:

```
--- SHOWING ERROR LOG FOR deck-1.25.0 ----------------------

this path will be fetched (0.01 MiB download, 0.03 MiB unpacked):
  /nix/store/5w53jaz6408yak154x0qii1xchmknjlv-common-updater-scripts
copying path '/nix/store/5w53jaz6408yak154x0qii1xchmknjlv-common-updater-scripts' from 'https://cache.nixos.org'...
error: cannot evaluate a function that has an argument without a value ('buildGoModule')
       Nix attempted to evaluate a function as a top level expression; in
       this case it must have its arguments supplied either by default
       values, or passed explicitly with '--arg' or '--argstr'. See
       https://nixos.org/manual/nix/stable/language/constructs.html#functions.

       at /home/joerg/git/nixpkgs/pkgs/applications/networking/deck/default.nix:1:3:

            1| { buildGoModule, lib, installShellFiles, fetchFromGitHub }:
             |   ^
            2| let
error: cannot evaluate a function that has an argument without a value ('buildGoModule')
       Nix attempted to evaluate a function as a top level expression; in
       this case it must have its arguments supplied either by default
       values, or passed explicitly with '--arg' or '--argstr'. See
       https://nixos.org/manual/nix/stable/language/constructs.html#functions.

       at /home/joerg/git/nixpkgs/pkgs/applications/networking/deck/default.nix:1:3:

            1| { buildGoModule, lib, installShellFiles, fetchFromGitHub }:
             |   ^
            2| let
error: cannot evaluate a function that has an argument without a value ('buildGoModule')
       Nix attempted to evaluate a function as a top level expression; in
       this case it must have its arguments supplied either by default
       values, or passed explicitly with '--arg' or '--argstr'. See
       https://nixos.org/manual/nix/stable/language/constructs.html#functions.

       at /home/joerg/git/nixpkgs/pkgs/applications/networking/deck/default.nix:1:3:

            1| { buildGoModule, lib, installShellFiles, fetchFromGitHub }:
             |   ^
            2| let
update-source-version: error: Could not find attribute 'deck'!


--- SHOWING ERROR LOG FOR deck-1.25.0 ----------------------
The update script for deck-1.25.0 failed with exit code 1
Traceback (most recent call last):
  File "/home/joerg/git/nix-update/bin/nix-update", line 12, in <module>
    main()
  File "/home/joerg/git/nix-update/nix_update/__init__.py", line 287, in main
    package = update(options)
              ^^^^^^^^^^^^^^^
  File "/home/joerg/git/nix-update/nix_update/update.py", line 321, in update
    run(
  File "/home/joerg/git/nix-update/nix_update/utils.py", line 36, in run
    return subprocess.run(
           ^^^^^^^^^^^^^^^
  File "/nix/store/9hwaihb8rhlvq8s7xq11m204jp51h6xi-python3-3.11.4/lib/python3.11/subprocess.py", line 571, in run
    raise CalledProcessError(retcode, process.args,
subprocess.CalledProcessError: Command '['nix-shell', '/home/joerg/git/nixpkgs/maintainers/scripts/update.nix', '--argstr', 'package', 'deck']' returned non-zero exit status 1.

```

Reply:
Thanks for telling me `nix-update --build --commit --use-update-script deck`. This helps a lot.

The script should work now, can you please have a look. 

Thanks